test/statement/where.js

```diff
@@ -151,6 +151,25 @@ describe('WHERE clause generation', function () {
         assert.equal(result.params[3], 'value4');
       });
 
+      it('should encapsulate and AND together inner conjunction subgroups', function () {
+        const result = where({
+          and: [{
+            field1: 'value1'
+          }, {
+            field2: 'value2', field3: 'value3'
+          }, {
+            field4: 'value4'
+          }]
+        });
+
+        assert.equal(result.conditions, '(("field1" = $1) AND ("field2" = $2 AND "field3" = $3) AND ("field4" = $4))');
+        assert.equal(result.params.length, 4);
+        assert.equal(result.params[0], 'value1');
+        assert.equal(result.params[1], 'value2');
+        assert.equal(result.params[2], 'value3');
+        assert.equal(result.params[3], 'value4');
+      });
```

Comment:
This is a good start but if you look at the next few testcases in `describe('with subgroups')`, there are three more validating disjunctions ('should not pollute other fields', 'should return a usable predicate if only given one subgroup', and 'recurses'). I don't think the 'one subgroup' test is necessary but the others would be really useful, especially 'recurses'.

It might be clearest if you were to copy+paste the `describe` and separate conjunctions and disjunctions that way.

Reply:
There, just did exactly that :)